cpp/include/raft/random/rmat_rectangular_generator.cuh

```diff
@@ -65,6 +68,58 @@ namespace raft::random {
  * @{
  */
 template <typename IdxT, typename ProbT>
+void rmat_rectangular_gen(const raft::handle_t& handle,
+                          raft::random::RngState& r,
+                          raft::device_vector_view<const ProbT, IdxT> theta,
+                          raft::device_vector_view<IdxT, IdxT> out,
+                          raft::device_vector_view<IdxT, IdxT> out_src,
```

Comment:
This output (and the one below) are optional since it's already included in `out`.

Reply:
I've consolidated the output vectors into a class that makes the meaning of the output vectors more clear, validates sizes, and prevents users from getting into an unsupported or incorrect state.  Thanks!